exporter/exporterhelper/internal/bounded_memory_queue_test.go

```diff
@@ -26,55 +24,45 @@ import (
 func TestBoundedQueue(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](1)
 
-	waitCh := make(chan struct{})
-
-	consumerState := newConsumerState(t)
-
-	consumers := NewQueueConsumers(q, 1, func(_ context.Context, item string) error {
-		consumerState.record(item)
-		<-waitCh
-		return nil
-	})
-	assert.NoError(t, consumers.Start(context.Background(), componenttest.NewNopHost()))
-
 	assert.NoError(t, q.Offer(context.Background(), "a"))
 
-	// at this point "a" may or may not have been received by the consumer go-routine
-	// so let's make sure it has been
-	consumerState.waitToConsumeOnce()
-
-	// at this point the item must have been read off the queue, but the consumer is blocked
+	numConsumed := 0
+	assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+		assert.Equal(t, "a", item)
+		numConsumed++
+		return nil
+	}))
+	assert.EqualValues(t, 1, numConsumed)
```

Comment:
Same question for the checks in this test

Reply:
No, Consume is sync.